komodo-core/src/main/java/org/komodo/repository/RepositoryImpl.java

```diff
@@ -433,31 +449,122 @@ public String decode(String encoded) {
+    private static final String WORKSPACE_ROOT = (KOMODO_ROOT + FORWARD_SLASH + Komodo.WORKSPACE);
 
     /**
-     * The root path of the Komodo repository workspace searches area
+     * The root path of the Komodo repository workspace searches area.
+     * This remains a sibling of the user home directories so that searches can be shared by users.
      */
     public static final String SEARCHES_ROOT = WORKSPACE_ROOT + FORWARD_SLASH + Search.GROUP_NODE;
 
     /**
-     * The root path of the Komodo repository environment validation rules area
+     * @param transaction
+     *       the transaction (cannot be <code>null</code> or have a state that is not
+    *        {@link org.komodo.spi.repository.Repository.UnitOfWork.State#NOT_STARTED})
+    *
+     * @return true if the given transaction is a system transaction
      */
-    public static final String VALIDATION_ROOT = ENV_ROOT + FORWARD_SLASH + Environment.VALIDATION;
+    public static boolean isSystemTx(UnitOfWork transaction) {
+        if (transaction == null)
+            return false;
+
+        //
+        // Transactions should always have a user name but just in case one sneaked through
+        //
+        return transaction.getUserName() == null || SYSTEM_USER.equals(transaction.getUserName());
+    }
 
     /**
-     * The root path of the Komodo repository environment servers area
+     * The komodo user's workspace in the repository, ie. /tko:komodo/tko:workspace/${user}
+     * where ${user} is the user owning the given transaction
+     *
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not
+     *        {@link org.komodo.spi.repository.Repository.UnitOfWork.State#NOT_STARTED})
+     *
+     * @return the workspace path for the user who owns the transaction
      */
-    public static final String SERVERS_ROOT = ENV_ROOT + FORWARD_SLASH + Environment.SERVERS;
+    public static String komodoWorkspacePath(final UnitOfWork uow) {
+        if(uow == null)
+            return WORKSPACE_ROOT;
+
+        String userName = uow.getUserName();
+        if (userName == null || isSystemTx(uow))
+            return WORKSPACE_ROOT;
+
+        return WORKSPACE_ROOT + FORWARD_SLASH + userName;
+    }
 
     /**
-     * The root path of the Komodo repository teiid cache area.
+     * @param path the path to test
+     *
+     * @return true if the path is a reserved path
      */
-    public static final String TEIID_CACHE_ROOT = ENV_ROOT + FORWARD_SLASH + Environment.TEIID_CACHE;
+    public static boolean isReservedPath(String path) {
+        if (path == null)
+            return false;
 
-    protected static final KLog LOGGER = KLog.getLogger();
+        //
+        // Ensure path has no trailing slash
+        //
+        if (path.endsWith(FORWARD_SLASH))
+            path = path.substring(0, path.length() - 1);
+
+        if (KOMODO_ROOT.equals(path) || LIBRARY_ROOT.equals(path) || ENV_ROOT.equals(path)
```

Comment:
Unfortunately, I cannot since getReservedPaths() requires a transaction, which in turn requires a repository. Since this method is static, I cannot get a repository.